python_modules/libraries/dagster-aws/dagster_aws_tests/secretsmanager_tests/test_secrets.py

```diff
@@ -0,0 +1,118 @@
+        Name="bar_secret", SecretString="bar_value", Tags=[{"Key": "airflow", "Value": "bar"}]
+    )
+    assert get_tagged_secrets(mock_secretsmanager_resource, "dagster") == {
+        "foo_secret": foo_secret["ARN"]
+    }
+
+    baz_secret = mock_secretsmanager_resource.create_secret(
+        Name="baz_secret", SecretString="baz_value", Tags=[{"Key": "dagster", "Value": "baz"}]
+    )
+    assert get_tagged_secrets(mock_secretsmanager_resource, "dagster") == {
+        "foo_secret": foo_secret["ARN"],
+        "baz_secret": baz_secret["ARN"],
+    }
+
+
+def test_secretmanager_secrets_resource(mock_secretsmanager_resource):
+    foo_secret = mock_secretsmanager_resource.create_secret(
+        Name="foo_secret", SecretString="foo_value", Tags=[{"Key": "dagster", "Value": "foo"}]
+    )
+    bar_secret = mock_secretsmanager_resource.create_secret(
+        Name="bar_secret", SecretString="bar_value", Tags=[{"Key": "airflow", "Value": "bar"}]
+    )
+    _baz_secret = mock_secretsmanager_resource.create_secret(
+        Name="baz_secret", SecretString="baz_value", Tags=[{"Key": "dagster", "Value": "baz"}]
+    )
+    asdf_secret = mock_secretsmanager_resource.create_secret(
+        Name="asdf_secret", SecretString="asdf_value"
+    )
+
+    with secretsmanager_secrets_resource(
+        build_init_resource_context(config={"secrets": [foo_secret["ARN"]]})
+    ) as secret_map:
+        assert secret_map == {"foo_secret": "foo_value"}
+
+    with secretsmanager_secrets_resource(
+        build_init_resource_context(config={"secrets": [foo_secret["ARN"], bar_secret["ARN"]]})
+    ) as secret_map:
+        assert secret_map == {"foo_secret": "foo_value", "bar_secret": "bar_value"}
+
+    with secretsmanager_secrets_resource(
+        build_init_resource_context(config={"secrets_tag": "dagster"})
+    ) as secret_map:
+        assert secret_map == {"foo_secret": "foo_value", "baz_secret": "baz_value"}
+
+    with secretsmanager_secrets_resource(
+        build_init_resource_context(
+            config={"secrets_tag": "dagster", "secrets": [asdf_secret["ARN"]]}
+        )
+    ) as secret_map:
+        assert secret_map == {
+            "foo_secret": "foo_value",
+            "baz_secret": "baz_value",
+            "asdf_secret": "asdf_value",
+        }
+
+        assert os.getenv("foo_secret") is None
+        assert os.getenv("baz_secret") is None
+        assert os.getenv("asdf_secret") is None
+
+    with secretsmanager_secrets_resource(
+        build_init_resource_context(
+            config={
+                "secrets_tag": "dagster",
+                "secrets": [asdf_secret["ARN"]],
+                "add_to_environment": True,
+            }
+        )
+    ) as secret_map:
+        assert secret_map == {
+            "foo_secret": "foo_value",
+            "baz_secret": "baz_value",
+            "asdf_secret": "asdf_value",
+        }
+
+        assert os.getenv("foo_secret") == "foo_value"
```

Comment:
Added tests for these three cases:

- Existing env vars are overwritten
- The value is `None`, since that's what the `SecretString` property of the secret is